src/mpi/init/initthread.c

```diff
@@ -96,6 +96,9 @@ MPIR_Per_thread_t MPIR_Per_thread = { 0 };
 
 MPID_Thread_tls_t MPIR_Per_thread_key;
 
+OPA_int_t num_server_thread = { 0 };
+MPIR_Thread_sync_list_t sync_wait_list = { NULL };
```

Comment:
It would be good to first document what purpose the server threads and wait list serve. That will make it easier to review.